### DISCUSSION TITLE
Jest Mocks Still Not Supported?

### DISCUSSION THREAD

**Discussion by Sayvai** [codesandbox/codesandbox-client#5042] [Q&A]
I've come across the issue where it seems Jest mocks are not supported. Where in my particular CodeSandbox [example](https://codesandbox.io/s/react-testing-library-example-p82ty?file=/src/__tests__/custom-component.test.js), when running the test, an exception is thrown in the test output `TypeError: jest.mock is not a function`.

I found an open issue (#513) which also describes the same problem, dating as far back as 2018. 

I then also stumbled across this archived Spectrum [chat](https://spectrum.chat/codesandbox/help/jest-mock-is-not-a-function~111527b2-0dfa-423e-ac42-312c296ec736) which confirmed Jest mocks are not supported, dating back to 2019.

Mocking is pretty much an integral part of writing unit tests.

Will Jest mocks ever be supported on the CodeSandbox client?

**Comment by throne1986** [comment]
It's November 19 2020 still having a corona problem at the same time I am still having the same problem I had in 2018 jest.mock is not a function !!!!!!  please do something

**Reply by Darren-Corbett** [reply]
still a problem

**Reply by berci-i** [reply]
04 April 2022 -> Still an issue.

**Comment by Sayvai** [comment]
Just come across this GitHub issue [comment](https://github.com/codesandbox/codesandbox-client/issues/513#issuecomment-834203439) from a CodeSandbox member:

>Yes, this is still something we want to do, but it's challenging. It's something that we haven't planned in the near-future, but we do want to expand the Jest functionality in the future. I can't tell when we'll have this though.

So it appears the team are aware of the issue, but it's proving to be a bit of a challenge, and so the team are unable to provide an estimate for the fix for now. Re-assuring to know that it's on their minds.
